### ISSUE TITLE
[Web/dom] `Style` + `if` make recompositions be appended rather than replaced

### ISSUE THREAD

**Issue by ptitjes** [JetBrains/compose-multiplatform#2568]
When using a stylesheet and an `if` statement, it seems that the recomposed content of the `if` statement is prepended to the existing content rather replacing it.

```kotlin
import androidx.compose.runtime.*
import org.jetbrains.compose.web.*
import org.jetbrains.compose.web.css.*
import org.jetbrains.compose.web.dom.*

fun startApp() {
    renderComposable(rootElementId = "root") {
        Style(AppStylesheet)
        Test()
    }
}

@Composable
fun Test() {
    var counter by remember { mutableStateOf(0) }

    if (counter % 2 == 0) {
        P({ classes(AppStylesheet.counter) }) {
            Text("$counter < Even")
        }
    } else {
        P({ classes(AppStylesheet.counter) }) {
            Text("$counter < Odd")
        }
    }

    Button({ onClick { counter++ } }) {
        Text("Increment")
    }
}

object AppStylesheet : StyleSheet() {
    val counter by style {
        fontWeight("bold")
    }
}
```

[Screencast from 2022-12-19 18-13-40.webm](https://user-images.githubusercontent.com/64956/208489483-64c5df6a-2cdc-4f2a-9acd-ea1151d0ee38.webm)

Tested on Kotlin 1.7.20 + Compose Multiplatform 1.2.2 (no compiler override).

**Comment by eymar** [comment]
@ptitjes Let's try to find more details.

I tried to reproduce the issue, but no luck so far. Here is a project I created with the code thar you provided https://github.com/eymar/repr_compose_2568

I even used 
`//compose.kotlinCompilerPlugin.set("androidx.compose.compiler:compiler:1.4.0-alpha02")`
and
`#kotlin.version=1.7.21`
and the behaviour was correct (no append)ds)

Perhaps you could try [this project](https://github.com/eymar/repr_compose_2568) and share your example on github? I'm wondering what's the difference.

**Comment by ptitjes** [comment]
OK. I will try it this evening. Thanks a lot for creating the project.

**Comment by eymar** [comment]
@ptitjes Did you have a chance to try it? :)

**Comment by ptitjes** [comment]
@eymar I am very sorry, I did not find time yet. I will try this weekend.

**Comment by alexzhirkevich** [comment]
Same issue was with Kotlin 1.7.20 + Compose 1.2.1. 

Solved by updating compose to 1.2.2

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2568">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
